qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -66,67 +103,96 @@ def copy(self):
+        **options,
+    ) -> ExperimentData:
+        # Make a new copy of experiment data if not updating results
+        if not replace_results and _requires_copy(experiment_data):
+            experiment_data = experiment_data.copy()
+
+        if not self.options.combine_results:
+            # Initialize child components if they are not initalized.
+            self._initialize_child_data(experiment_data)
+
+        # Run analysis with replace_results = True since we have already
+        # created the copy if it was required
+        return super().run(experiment_data, replace_results=True, **options)
+
     def _run_analysis(self, experiment_data: ExperimentData):
         # Return list of experiment data containers for each component experiment
         # containing the marginalied data from the composite experiment
-        component_exp_data = self._component_experiment_data(experiment_data)
+        component_expdata = self._component_experiment_data(experiment_data)
 
         # Run the component analysis on each component data
-        for sub_exp_data, sub_analysis in zip(component_exp_data, self._analyses):
+        for i, sub_expdata in enumerate(component_expdata):
             # Since copy for replace result is handled at the parent level
             # we always run with replace result on component analysis
-            sub_analysis.run(sub_exp_data, replace_results=True)
+            self._analyses[i].run(sub_expdata, replace_results=True)
+
+            if not self.options.run_parallel:
+                # Block for current components analysis to finish before
+                # starting the next components analysis
+                sub_expdata.block_for_results()
+
+        if self.options.run_parallel:
+            # Analysis is running in parallel so we add loop to wait
+            # for all component analysis to finish before returning
+            # the parent experiment analysis results
+            for sub_expdata in component_expdata:
+                sub_expdata.block_for_results()
 
-        # Wait for all component analysis to finish before returning
-        # the parent experiment analysis results
-        for sub_exp_data in component_exp_data:
-            sub_exp_data.block_for_results()
+        analysis_results = []
+        figures = []
+        if self.options.combine_results:
+            # Optionally combine results into main container
+            for i, sub_expdata in enumerate(component_expdata):
+                figures += sub_expdata._figures.values()
+                for result in sub_expdata.analysis_results():
+                    # Add metadata to distinguish the component experiment
+                    # the result was generated from
+                    result.extra["component_experiment"] = {
+                        "experiment_type": sub_expdata.experiment_type,
+                        "component_index": i,
+                    }
+                    analysis_results.append(result)
```

Comment:
My intent was you can't ever save the component experiment data in that case, since it only ever exists as a temporary during analysis. However i think you've pointed out a bug that in this case it should always have auto-save set to False so you can't accidently save it.